### PR TITLE
fix: add json content to flow api

### PR DESCRIPTION
**Pull request by Bars92**
## Related Issues

related https://github.com/descope/etc/issues/6880

## Related PRs
https://github.com/descope/consoleservice/pull/2026

## Description
- fix content type for flow start and next routes

## Must

- [ ] Tests
- [ ] Documentation (if applicable)

### CODE REVIEW

**Comment by asafshen** [comment]
I think it will be better to handle this in the receiving side if its possible,

if not, please write why, and make sure that this is communicated throughout the code comments, as future reader will not know why those APIs are the only one that needs this

**Review by orius123 — DISMISSED** [review]
@nirgur @dorsha do we want to it for all APIs?

**Comment by Bars92** [comment]
@asafshen its not possible. this is the content type of the request itself. sadly we haven't changed it and default is plain text which proto ignores. some apis may need this as well

**Comment by asafshen** [comment]
> @asafshen its not possible. this is the content type of the request itself. sadly we haven't changed it and default is plain text which proto ignores. some POST apis may need this as well - they are all addressing to us so it probably doesn't really matter

noting that the fact that we can avoid this header so far improved simplicity (and even a bit of performance)

I'm not sure if its possible, but I would do the extra mile and attempty handle thiw in the internal service (even as plain text), rather then introducing this complexity here, but this current solution is also ok I guess, worth to add explanation

**Comment by Bars92** [comment]
@asafshen ill add explanation but i rather do this which will fix the incorrect content type we are trying to use and avoid ignoring content type in console service to avoid security implications

**Comment by asafshen** [comment]
cool, 

plus one for changing this in all places then 👍

**Comment by Bars92** [comment]
@asafshen done

File: packages/core-js-sdk/src/httpClient/index.ts [hunk 1]
```diff
@@ -36,6 +40,16 @@ const createDescopeHeaders = () => {
   };
 };
 
+const isJson = (value?: string) => {
+  try {
+    value = JSON.parse(value);
```
Comment (asafshen) [inline packages/core-js-sdk/src/httpClient/index.ts:45]: What's the harm of adding it always?

**Review by asafshen — DISMISSED** [review]

File: packages/core-js-sdk/src/httpClient/index.ts (hunk 1 above)
Comment (Bars92) [reply packages/core-js-sdk/src/httpClient/index.ts:45]: its usually bad practice on non POST/PUT and sometimes they may not send json body. i wouldn't want to change anything

**Review by asafshen — APPROVED** [review]
👍

**Pull request merged**